### ISSUE TITLE
When saving a state, the only feedback is that a PNG is saved

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1812]
This is very confusing for most users not familiar with openMSX. The only feedback is about saving a PNG, "that can't be right!" "Is the savestate stored in the PNG?".

Perhaps we should simply suppress the PNG saving message when saving a state (somehow). A feedback that the state was saved could also be good of course.

**Comment by m9710797** [comment]
I agree. The feedback is both confusing and superfluous. E.g. if you select "File > Save" in a typical application you also don't get an on-screen (or status bar, ...) message that the file was saved.

I think the original idea behind the message was for the use-case that the `screenshot` command was bound to the `print-screen` shortcut. When pressing shortcuts it can be useful to get some kind of feedback.

Maybe we should remove the message from the `screenshot` command itself, and instead add it to the default shortcut.

**Comment by MBilderbeek** [comment]
Yep, that could work.

Op za 14 dec. 2024 09:30 schreef Wouter Vermaelen ***@***.***

**Comment by m9710797** [comment]
Implemented in f62772f66d6bea51e10a3115c4e9e0fbfcfe0c4e. But it would be good if someone else could verify. E.g. there's now no feedback anymore (via a message) that the savestate was created.

**Comment by MBilderbeek** [comment]
I added some tweaks. Feel free to tweak more or revert if you disagree completely :)

**Comment by m9710797** [comment]
Thanks! I agree with your changes.

I think for the future we need to follow these guidelines:
* console commands should (usually) not generate messages. Because that (in some cases) makes them not usable from within a larger Tcl script. And when typed interactively on the console you don't need that message: you'll either get an error in the console output, or not (sometimes no output), and then you know it was successful.
* For GUI actions a message can sometimes be useful. And that's the approach you took in d53106535698ac30bb96155796510e101f0d58da.
* And (especially?) for hotkeys a message may be useful. But then we can encode that in the binding itself. See the example below.

A counter example is the screenshot command. That one still emits a message from the command itself. A better approach could be to remove that, and then change the hotkey binding to something like `bind PrintScreen {message "Screenshot saved to: [screenshot -guess-name]"}`

**Comment by MBilderbeek** [comment]
Glad you like it. I agree with the screenshot, but as discussed on IRC, there's a bit of a problem to implement it for all cases, as the "no sprites" option is asynchronous, so we can't return a value and use that as feedback and trigger to know about success/failure as in the other cases.

**Comment by MBilderbeek** [comment]
Original problem solved.
